services/wireguard/connection/connection.go

```diff
@@ -54,35 +58,37 @@ func (c *Connection) Start(options connection.ConnectOptions) (err error) {
 		return err
 	}
 
-	c.connection.Add(1)
+	c.ctx, c.cancel = context.WithCancel(context.Background())
```

Comment:
I am not sure context is really needed here as it was introduced for different purposes. Actually there are alot of dicusssions around this topic in golang community. My humble opinion - if it can be avoided - don't use it. I guess simple channel will do the trick anyway.

Reply:
Replaced context by the channel.